src/main/java/nl/andrewl/email_indexer/data/export/query/QueryExporterParams.java

```diff
@@ -0,0 +1,85 @@
+package nl.andrewl.email_indexer.data.export.query;
+
+import java.nio.file.Path;
+import java.util.List;
+import java.util.Objects;
+
+import nl.andrewl.email_indexer.data.EmailDataset;
+import nl.andrewl.email_indexer.data.EmailEntryPreview;
+import nl.andrewl.email_indexer.data.EmailRepository;
+import nl.andrewl.email_indexer.data.TagRepository;
+
+/**
+ * Parameter object for the EmailDatasetExporter.
+ */
+public record QueryExporterParams(String query, EmailRepository repository, TagRepository tagRepository,
+        EmailDataset dataset, Path directory, int maxResultCount, String fileType, boolean separateThreads,
+        List<EmailEntryPreview> emails) {
```

Comment:
Regarding the structure of this, I think it would be cleaner to have `QueryExporter` extend from the existing `EmailDatasetExporter` (or even make it become an abstract class implementation of it), rather than have two completely separate implementations with their own style.

As a second point, I don't think that it is a good idea to pre-fetch even just the email previews, as there may be tens of thousands, and as I've encountered, Mohamed's laptop is pretty slow and I've had to lazily-load everything in order to make things work.

Reply:
I just resolved the first half of this comment. 

I'm not sure if and how I should go about the second half though. Since the number of mails is capped, this shouldn't be much of an issue, I think?